### PR TITLE
[#162084] Change model association for DurationRate

### PR DESCRIPTION
**Pull request by LeticiaErrandonea**
# Release Notes
[Tech task] Update DB so Stepped billing rates can be validated agains Price policy usage rate

### CODE REVIEW

File: app/models/rate_start.rb [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 class RateStart < ApplicationRecord
 
-  has_many :duration_rates
+  has_many :duration_rates, dependent: :destroy
```
Comment (LeticiaErrandonea) [inline app/models/rate_start.rb:5]: This change doesn't actually belong here, but it's needed. I can undo it and add this in the next PR

File: db/migrate/20231108210613_change_associations_duration_rates.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+class ChangeAssociationsDurationRates < ActiveRecord::Migration[7.0]
```
Comment (LeticiaErrandonea) [inline db/migrate/20231108210613_change_associations_duration_rates.rb:1]: Do we want to do something about existing DurationRates? At least destroy them all?

File: app/models/rate_start.rb (hunk 1 above)
Comment (giladshanan) [reply app/models/rate_start.rb:5]: Smaller PRs are good - if this isn't related to model changes, save it for later

File: app/services/price_policy_updater.rb [hunk 3]
```diff
@@ -126,15 +126,12 @@ def permitted_params
...
-          :id,
-          :subsidy,
-          :rate,
-          :price_group_id,
-          :rate_start_id
-        ]
+        :subsidy,
+        :rate,
+        :price_policy_id,
+        :rate_start_id
```
Comment (giladshanan) [inline app/services/price_policy_updater.rb:134]: This feels right

File: db/migrate/20231108210613_change_associations_duration_rates.rb (hunk 2 above)
Comment (giladshanan) [reply db/migrate/20231108210613_change_associations_duration_rates.rb:1]: 💣  Yes let's start fresh

**Review by giladshanan — COMMENTED** [review]
Looks good so far - this feels like the right direction

File: db/migrate/20231108210613_change_associations_duration_rates.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
```
Comment (giladshanan) [inline db/migrate/20231108210613_change_associations_duration_rates.rb:1]: Please add frozen string literal magic comment

File: db/migrate/20231108210613_change_associations_duration_rates.rb [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+  def up
+    DurationRate.destroy_all
```
Comment (giladshanan) [inline db/migrate/20231108210613_change_associations_duration_rates.rb:3]: Can you use either SQL or define `DurationRate` in the migration file?  Just to be [safe](https://jakeyesbeck.com/2021/04/10/avoid-models-in-migrations/)

File: db/migrate/20231108210613_change_associations_duration_rates.rb (hunk 4 above)
Comment (LeticiaErrandonea) [reply db/migrate/20231108210613_change_associations_duration_rates.rb:3]: I'll have to add a sticky note somewhere so I don't keep forgetting to do it 🤦‍♀️

**Review by giladshanan — APPROVED** [review]

File: db/migrate/20231108210613_change_associations_duration_rates.rb (hunk 4 above)
Comment (giladshanan) [reply db/migrate/20231108210613_change_associations_duration_rates.rb:3]: we should set up rubocop to help us remember things like this

**Pull request merged**